contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,933 @@
+     * configured by the `poolWethShare` property. This function will respect liquidity
+     * ratios when there no liquidity yet in the pool. If liquidity is already present
+     * then it relies on the `_swapToDesiredPosition` function in a step before to already
+     * move the trading price to desired position (with some tolerance).
+     */
+    // rebalance already has re-entrency checks
+    // slither-disable-start reentrancy-no-eth
+    function _addLiquidity() internal {
+        uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
+        uint256 _oethbBalance = IERC20(OETHb).balanceOf(address(this));
+        require(_wethBalance > 0, "Must add some WETH");
+
+        uint160 _currentPrice = getPoolX96Price();
+        // sanity check active trading price is positioned within our desired tick
+        if (_currentPrice <= sqrtRatioX96Tick0 ||
+                _currentPrice >= sqrtRatioX96Tick1) {
+            int24 currentTick = getCurrentTradingTick();
+            revert OutsideExpectedTickRange(
+                currentTick,
+                lowerTick,
+                upperTick
+            );
+        }
+
+        // in case oethb would be the 1st token we'd need to call estimateAmount0 here
+        uint256 _oethbRequired = helper.estimateAmount1(
+            _wethBalance,
+            address(clPool),
+            _currentPrice,
+            lowerTick,
+            upperTick
+        );
+
+        if (_oethbRequired > _oethbBalance) {
+            IVault(vaultAddress).mintForStrategy(
+                _oethbRequired - _oethbBalance
+            );
+        }
+
+        uint256 _wethAmountSupplied;
+        uint256 _oethbAmountSupplied;
+        if (tokenId == 0) {
+            (
+                uint256 _mintedTokenId,
+                ,
+                uint256 __wethAmountSupplied,
+                uint256 __oethbAmountSupplied
+            ) = positionManager.mint(
+                    /** amount0Min & amount1Min are left at 0 because slippage protection is ensured by the
+                     * _checkForExpectedPoolPrice
+                     *
+                     * Also sqrtPriceX96 is 0 because the pool is already created
+                     * non zero amount attempts to create a new instance of the pool
+                     */
+                    INonfungiblePositionManager.MintParams({
+                        token0: WETH,
+                        token1: OETHb,
+                        tickSpacing: tickSpacing,
+                        tickLower: lowerTick,
+                        tickUpper: upperTick,
+                        amount0Desired: _wethBalance,
+                        amount1Desired: _oethbRequired,
+                        amount0Min: 0,
+                        amount1Min: 0,
+                        recipient: address(this),
+                        deadline: block.timestamp,
+                        sqrtPriceX96: 0
```

Comment:
⚪️ I know why sqrtPriceX96 should be zero, and you know why it should be zero, but let's add a comment so the rest of the world knows too.

Reply:
<img width="894" alt="Screenshot 2024-08-23 at 12 14 44" src="https://github.com/user-attachments/assets/85c4af81-694e-4b74-8a40-d9e94d0a6760">

So the comment is already above the struct. I have issues placing the comments in the middle of the struct because it doesn't compile. Both forms below would fail compilation: 
<img width="909" alt="Screenshot 2024-08-23 at 12 16 16" src="https://github.com/user-attachments/assets/c1119f19-4abf-424a-b869-e11a2b9c2d98">
<img width="860" alt="Screenshot 2024-08-23 at 12 15 44" src="https://github.com/user-attachments/assets/8c00e6a1-60f5-4c97-937d-84ae81ef0f0c">

Reply:
Cool. My bad. We're good here there.

Reply:
I was surprised too that it wouldn't compile